dulwich/tests/test_repository.py

```diff
@@ -521,6 +521,23 @@ def check(repo):
             check(nonbare)
             check(bare)
 
+    def test_working_tree(self):
+        temp_dir = tempfile.mkdtemp()
```

Comment:
Please create the new repository from scratch, don't rely on testdata. You can easily do this using Repo.init.